core/runtime.rs

```diff
@@ -163,6 +172,25 @@ pub(crate) struct JsRuntimeState {
   pub(crate) shared_array_buffer_store: Option<SharedArrayBufferStore>,
   pub(crate) compiled_wasm_module_store: Option<CompiledWasmModuleStore>,
   waker: AtomicWaker,
+
+  #[cfg(feature = "napi")]
```

Comment:
All of these `cfg` attributes are painful, maybe we could store all of the related NAPI fields in a helper struct?